src/select/Menu.js

```diff
@@ -86,47 +106,29 @@ class Menu extends Component {
 
   render() {
     const {
-      allOption,
-      allowSelectAll,
-      focusedOption,
-      getOptionLabel,
-      labelKey,
       menuRef,
-      onOptionClick,
-      onOptionFocus,
-      optionComponent,
-      optionRef,
-      optionRenderer,
       options,
-      staticOption,
-      value,
-      valueKey
+      staticOption
     } = this.props
 
-    const OptionComponent = optionComponent
-    const selected = staticOption && isSelected(staticOption, value, valueKey)
-    const isFocused = staticOption && staticOption === focusedOption
+    if (staticOption) {
+      const isStaticOptionRendered =
+        options.find(o => o.abbreviation === staticOption.abbreviation)
+
+      if (isStaticOptionRendered) {
+        // not ideal to mutate this array...
+        // however, we already re-render the entire menu onFocus
+        // TODO: localize state once React is upgraded
+        options.splice(-1, 1)
```

Comment:
Why do we need to splice this?

Reply:
This is to enable changing the staticOption thru props. Before pushing the `staticOption` into `options`, we remove the previous version.

Otherwise, another staticOption gets rendered below it with every render of the menu.

Since the whole menu is being rendered, this shouldn't be any worse that what was previously there.

Reply:
This is the part that is definitely not ideal. I can try to find a way around it, but it might take more refactoring.

Reply:
I removed the splice line and tested it out. I didn't see any duplicate StaticOptions occur. Isn't `options` a prop thats passed all the way down? Do you know of cases when `options` would contain the `Static Option`?

I think you can simplify this to:

```
let menuOptions  = staticOption ? [...options, staticOption] : options
```

Reply:
The duplicates aren't being rendered, but React throws an error in the console: 

![Screen Shot 2020-05-28 at 9 27 20 AM](https://user-images.githubusercontent.com/50337013/83168082-d3c31380-a0c5-11ea-8185-c0f58d6ac8dc.png)

Every subsequent render will already contain `staticOption` because we're modifying the `options` array (which comes from props, hence not ideal). 

I tried making a deep copy of the props before rendering them, but that didn't jive with the focus/blur/select logic. 

That's why we need to splice the "old version" out before pushing the "new version" into our options. This works because we're re-rendering the entire menu, which in itself isn't ideal.

So, on the first render, we push the `staticOption`, and every subsequent render will need to "replace" `staticOption` rather than pushing it

Reply:
I think a more robust approach would be to combine `options` and `staticOptions` at a higher level. But that doesn't fit the current patterns.

Reply:
Based on our conversation, I am trying a different approach. Reverted away from this pattern.